onyo/lib/onyo.py

```diff
@@ -92,6 +92,94 @@ def clear_caches(self) -> None:
         self._gitfiles = None
         self._templates = None
 
+    def generate_commit_message(self, message: list[str] = [], cmd: str = "",
+                                keys: list[str] = [], destination: str = "") -> str:
+        """
+        If `message` is given, the function uses the first element of it to
+        generate a message subject, and the following ones for the message body.
+
+        If no `message` is given, generates a default commit message based on
+        the other parameters given, and the `files staged` in the repository.
+        The paths for the message are shortened if needed, so that the resulting
+        message subject does not exceed `MAX_LEN`.
+
+        Adds a list of `changed files` with there path relative to the root of
+        the repository to the body of all commit messages.
+        """
+        message_subject = ""
+        message_body = ""
+        message_appendix = ""
+        MAX_LEN = 80
+
+        # staged files and directories (without ".anchor") in alphabetical order
+        staged_changes = [Path(self.opdir, x).relative_to(self.root).parent
+                          if str(Path(self.opdir, x).relative_to(self.root).name) == ".anchor"
+                          else Path(self.opdir, x).relative_to(self.root)
+                          for x in sorted(self.files_staged)]
+
+        if message:
+            message_subject = message[0][0]
+            message_body = self._n_join([x[0] for x in message[1:]])
+        else:
+            # get variables for the begin of the commit message `msg_dummy`
+            dest = None
+            keys_str = ""
+            if keys:
+                keys_str = f" ({','.join(str(x.split('=')[0]) for x in sorted(keys))})"
+            if destination:
+                dest = Path(self.opdir, destination).relative_to(self.root)
+            if dest and str(dest.name) == ".anchor":
+                dest = dest.parent
+
+            # build the commit message subject
+            msg_dummy = f"{cmd} [{len(staged_changes)}]{keys_str}"
```

Comment:
This variable name is not intuitive to me. Why is it a "dummy"?

I think comments here would really help to explain what's going on. It seems to always reference keys, even if we're not using `set`, etc.

Reply:
Thanks, I will add comments to it.

Yes, I think always refering `keys` is the better way, because it saves the `if keys`, because we need to initialize keys anyways. I find it easier to read, and especially modify, the line this way, because otherwise we had just a bigger if/else tree with many similar lines, that all set keys.